bpf/bpf_host.c

```diff
@@ -1697,6 +1676,39 @@ int cil_to_host(struct __ctx_buff *ctx)
 	 * to mark as PACKET_OTHERHOST and drop.
 	 */
 	ctx_change_type(ctx, PACKET_HOST);
+#if !defined(TUNNEL_MODE)
+	/* Since v1.18 Cilium performs IPsec encryption at the native device,
+	 * before the packet leaves the host.
+	 *
+	 * A special case exists for L7 egress proxy packets when native routing
+	 * mode is enabled.
+	 *
+	 * Because L7 egress proxy packets are generated in the host-namespace
+	 * and generated packets MUST adjust their MTU for ESP encapsulation
+	 * an IP route MTU adjustment exists for L7 egress proxy packets.
+	 *
+	 * When the L7 egress proxy generates packets an 'ip rule' in the host
+	 * namespace routes these packets into table 2005 which has a route
+	 * toward 'cilium_host' and adjusts the MTU correctly for ESP encap.
+	 *
+	 * When 'cil_from_host@cilium_host' is reached the skb's mark is zeroed
+	 * and the packet is pushed toward 'cil_to_host@cilium_net'.
+	 *
+	 * If we simply let this packet drop to the stack, an iptables rule
+	 * exists which will mark the packet with 0x200 and trigger a local
+	 * delivery as part of L7 Proxy TPROXY mechanism.
+	 *
+	 * This iptables rule is ignored by the mark MARK_MAGIC_PROXY_TO_WORLD.
+	 * Technically, it is also ignored by MARK_MAGIC_ENCRYPT but reusing
+	 * this mark breaks further processing as its used in the XFRM subsystem.
+	 *
+	 * Therefore, if the packet's mark is zero, indicating it was forwarded
+	 * from 'cilium_host', mark the packet with MARK_MAGIC_PROXY_TO_WORLD
```

Comment:
This construction is certainly "fun" ;). My main concern would be that this applies on "both" sides of the veth pair (we attach `to-host` twice). Can we limit this to exactly where it's needed?

Reply:
Thanks for calling this out, it's been a concern of mine too. 

Essentially we only want this to run on `cilium_net`. 
We could check `THIS_INTERFACE_MAC == CILIUM_NET_MAC` check.

Reply:
> Essentially we only want this to run on `cilium_net`.
> We could check `THIS_INTERFACE_MAC == CILIUM_NET_MAC` check.

:+1: I think we typically use `THIS_INTERFACE_IFINDEX`, but same-same. Best double-check if the loader is already injecting those macros correctly before you dig too deep, this feels easy to miss.

Reply:
This is handled in latest commit. All my local tests are passing. Lets run it thru CI.